modules/mcuboot/CMakeLists.txt

```diff
@@ -6,6 +6,26 @@ if (CONFIG_MCUBOOT)
   if (CONFIG_NRF53_MULTI_IMAGE_UPDATE)
     set_shared(IMAGE ${IMAGE_NAME} PROPERTY NRF53_MULTI_IMAGE_UPDATE
       ${CONFIG_NRF53_MULTI_IMAGE_UPDATE})
+  if (NOT DEFINED CONFIG_BOOT_UPGRADE_ONLY)
+    # Added color defines due to this coming from a childe image and being out
+    # putted as raw unformatted text to increase visibility.
+    string(ASCII 27 Esc)
+    set(ColorReset "${Esc}[m")
+    set(BoldRed     "${Esc}[1;31m")
+    message(WARNING "${BoldRed}
+        ---------------------------------------------------------
+        --- WARNING: Using multi-image update on nRF53        ---
+        --- without BOOT_UPGRADE_ONLY corrupts the network    ---
+        --- upon image reverts. Only application core can be  ---
+        --- reverted.                                         ---
+        ---------------------------------------------------------
+	\n${ColorReset}"
+        )
```

Comment:
why is such a huge banner needed ?

Why not just a single `WARNING` line.
A single line of warning will still provide users with a stack trace, and should get their attention.

It's not like their device is being destroyed if they do this, they can still re-flash and one must expect our users to have examined any warnings carefully and tested their product before putting software into production.

A regular warning is sufficient here.

Reply:
> The commit message says:
> 
> > This requires you have a method of loading new network core data
> > into external flash from your application core application.
> 
> so how does the user informs the build system that such a method is in place ? Users should be able to inform the build system that they know what they are doing, and thereby silence the `warning`.

The build system currently does not, and have no way of disabling the warning good point.

> 
> Or simply have another prompt-less setting that users app can forcefully select:
> 
> ```
> depends on BOOT_UPGRADE_ONLY || I_KNOW_WHAT_IM_DOING_FLAG
> ```
> 
> then user's app `Kconfig` can do:
> 
> ```
> # <app>/Kconfig
> 
> config MY_APP
>     default y
>     select I_KNOW_WHAT_IM_DOING_FLAG
> ```

I think having some promtless setting is the way to go and have it documented in `.rts` instead. Thank you for the ide @tejlmand I'll refactor the PR.